### PR TITLE
Replace MathJax with KaTeX

### LINKED ISSUE #495: Replace React-MathJax 2 with KaTeX
**Is your feature request related to a problem? Please describe.**
We're using React-Mathjax, which uses MathJax 2 and doesn't support mathjax 3.
I can't find a react-lib that supports mathjax 3. How about switching to KaTeX?
KaTeX is [faster](https://www.intmath.com/cg5/katex-mathjax-comparison.php), can also process [TeX](https://katex.org/) and there are better maintained [react components](https://github.com/MatejBransky/react-katex)

**Describe the solution you'd like**
Use KaTeX

**Describe alternatives you've considered**
Create our own mathjax 3 component or use mathjax 2.

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Component/Part
Markdown renderer

### Description
This PR replaces mathjax with katex.

### Steps
- [x] added implementation
- [ ] added / updated tests
- [ ] added / updated documentation
- [x] extended changelog

### Related Issue(s)
Closes #495

### CODE REVIEW

**Comment by mrdrogdrog** [comment]
Changelog is still missing

**Comment by mrdrogdrog** [comment]
KaTeX is incompatible with node 10

**Comment by mrdrogdrog** [comment]
We (@codimd/frontend) decided, that it's the best to drop the node 10 support.

**Comment by ccoenen** [comment]
just to make this clear: this does not affect the server's node 10 compatibility. One can build this on node12 and still run it with a node 10 server, right?

**Comment by mrdrogdrog** [comment]
Yes. NodeJS is not needed to run the frontend, because it is compiled into static files.

File: src/components/markdown-renderer/markdown-renderer.tsx [hunk 1]
```diff
@@ -244,12 +243,12 @@ export const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, onM
       permalinkSymbol: '<i class="fa fa-link"></i>'
     })
     md.use(mathJax({
```
Comment (DerMolly) [inline src/components/markdown-renderer/markdown-renderer.tsx:245]: do we want to rename this component, too?

File: src/components/markdown-renderer/replace-components/katex/katex-replacer.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { DomElement } from 'domhandler'
+import React from 'react'
+import 'katex/dist/katex.min.css'
+import TeX from '@matejmazur/react-katex'
+import { ComponentReplacer } from '../ComponentReplacer'
+
+const getNodeIfMathJaxBlock = (node: DomElement): (DomElement|undefined) => {
```
Comment (DerMolly) [inline src/components/markdown-renderer/replace-components/katex/katex-replacer.tsx:7]: same here

**Review by DerMolly — CHANGES_REQUESTED** [review]

File: src/components/markdown-renderer/markdown-renderer.tsx (hunk 1 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/markdown-renderer.tsx:245]: We can rename the import, but not the lib.

File: src/components/markdown-renderer/replace-components/katex/katex-replacer.tsx (hunk 2 above)
Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/katex/katex-replacer.tsx]: good point.

Comment (mrdrogdrog) [reply src/components/markdown-renderer/replace-components/katex/katex-replacer.tsx]: solved

**Review by DerMolly — APPROVED** [review]

**Review by ErikMichelson — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -48,6 +48,7 @@
 - The `[name=...]`, `[time=...]` and `[color=...]` tags may now be used anywhere in the document and not just inside of blockquotes and lists.
 - The <i class="fa fa-picture-o"/> (add image) and <i class="fa fa-link"/> (add link) toolbar buttons, put selected links directly in the `()` instead of the `[]` part of the generated markdown
 - The help dialog has multiple tabs, and is a bit more organized.
+- Use KaTeX instead of MathJax.
```
Comment (ccoenen) [inline CHANGELOG.md:51]: someone posted a comparison link to the chat. Maybe this would be great information for someone who reads this change.

Comment (ccoenen) [reply CHANGELOG.md:51]: or perhaps this one (if it is not already somewhere else, maybe the features page would also be a good place for this)
https://katex.org/docs/supported.html

Comment (DerMolly) [reply CHANGELOG.md:51]: the features page is currently not in this repo

Comment (mrdrogdrog) [reply CHANGELOG.md]: I added a link to our FAQ page.

**Pull request merged**